custom/motioncontrollers/CustomPIDController.java

```diff
@@ -319,4 +344,9 @@ public double get() {
 			return 0;
 		}
 	}
+
+	@Override
+	public boolean onTarget() {
+		return super.onTarget() && derivativeTolerance != 0 && Math.abs(lastErrorDerivative) < derivativeTolerance;
```

Comment:
The derivative logic should be in its own function like onTarget()? I guess that's kind of what it is, but it seems kind of weird here.